library/ssl_tls.c

```diff
@@ -6529,7 +6529,7 @@ static void ssl_calc_finished_tls_sha256(
     unsigned char padbuf[32];
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
     size_t hash_size;
-    psa_hash_operation_t sha256_psa;
+    psa_hash_operation_t sha256_psa = PSA_HASH_OPERATION_INIT;
```

Comment:
Using `PSA_HASH_OPERATION_INIT`, `psa_hash_operation_init();` or simply `= 0` are all valid ways to do initialisation, right?

What are the advantages of different methods? In particular, why are we using on line 1449 and 1492 `psa_hash_operation_init();` and `PSA_HASH_OPERATION_INIT` here?

Reply:
`{0}` and `INIT` (macro) can be used to initialize global variables, but `init()` (function) can't. `{0}` is generic but some compilers warn that this implicitly initializes other fields to 0. `init()` is the only one that's typed, so you'll get an error if you use it in the wrong context. `memset` to 0 is also defined as equivalent because it allows applications to use e.g. `calloc`, and in practice it produces the same value as `{0}` on virtually all platforms.

In most cases, the choice is indifferent.

Reply:
`PSA_HASH_OPERATION_INIT` is all static, so you can use it for static globals. The inline function can be used where functions can be used, which is a few less places, but it uses the static initializer under-the-hood. `= {0}` is usually ok, but some compilers will complain about missing field initializers.

I personally prefer using the macro everywhere, but previous authors inside library/ssl_tls.c liked the function. There is no functional difference in this case.